include/Fader.h

```diff
@@ -100,14 +100,12 @@ class EXPORT Fader : public QWidget, public FloatModelView
 
 	int knobPosY() const
 	{
-		float fRange = m_model->maxValue() - m_model->minValue();
-		float realVal = m_model->value() - m_model->minValue();
+		float fRange = model()->maxValue() - model()->minValue();
+		float realVal = model()->value() - model()->minValue();
 
 		return height() - ( ( height() - m_knob->height() ) * ( realVal / fRange ) );
 	}
 
-	FloatModel * m_model;
```

Comment:
On 01/23/2015 05:30 PM, Amadeus Folego wrote:
>
> In include/Fader.h
> <https://github.com/LMMS/lmms/pull/1678#discussion_r23456242>:
>
> >  
> >  		return height() - ( ( height() - m_knob->height() ) * ( realVal / fRange ) );
> >  	}
> >  
> > -	FloatModel * m_model;
>
> As described on the PR. This was creating ambiguity since this widget
> is *already* an |Float ModelView|, so instead of accessing the inner
> variable |m_model| that is inherited from that you would have a bug
> where a different, out of the blue, model is accessed.
>
> This was the cause of hours and hours of debugging to try to
> understand why when I moved the fader model, it was *still* being
> modeled by the other FX Variable.
>
> This is fixed by using |model()|.
>

Cool, that seems sensible.